### PR TITLE
pass check if only tolerance is set

### LINKED ISSUE #665: setting tolerance instead of max in delay options doesn't work
Summary:
The `delay_opts()` family of functions sets a default tolerance and thus should work without setting a maximum. Currently, however, doing so results in an error.

Reproducible Steps:
``` r
library("EpiNow2")
#> 
#> Attaching package: 'EpiNow2'
#> The following object is masked from 'package:stats':
#> 
#>     Gamma
generation_time_opts(Gamma(mean = 3.6, sd = 3.1))
#> Error in check_stan_delay(dist): All distribution passed to the model need to have a finite maximum
```

<sup>Created on 2024-05-09 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

Related: #533

### PR DESCRIPTION
**Pull request by sbfnk**
## Description

This PR closes #665.

## Initial submission checklist

- [X] My PR is based on a package issue and I have explicitly linked it.
- [X] I have tested my changes locally (using `devtools::test()` and `devtools::check()`).
- [X] I have added or updated unit tests where necessary.
- [X] I have updated the documentation if required and rebuilt docs if yes (using `devtools::document()`).
- [X] I have followed the established coding standards (and checked using `lintr::lint_package()`).
- [X] I have added a news item linked to this PR.

## After the initial Pull Request 

- [x] I have reviewed Checks for this PR and addressed any issues as far as I am able.

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -10,7 +10,8 @@
 
 ## Bug fixes
 
-- a bug was fixed where `forecast_secondary()` did not work with fixed delays. By @sbfnk in #717 and reviewed by @sbfnk.
+- a bug was fixed that caused delay option functions to report an error if only the tolerance was specified. By @sbfnk.
```
Comment (jamesmbaazam) [inline NEWS.md:13]:
```suggestion
- a bug was fixed that caused delay option functions to report an error if only the tolerance was specified. By @sbfnk in #716 and reviewed by @jamesmbaazam.
```

File: R/checks.R [hunk 2]
```diff
@@ -96,8 +96,9 @@ check_stan_delay <- function(dist) {
       "and infinite maximum."
     )
   }
+  assert_numeric(attr(dist, "tolerance"), lower = 0, upper = 1)
   # Check that `dist` has a finite maximum
-  if (any(is.infinite(max(dist)))) {
+  if (any(is.infinite(max(dist))) && !(attr(dist, "tolerance") > 0)) {
     stop("All distribution passed to the model need to have a finite maximum")
```
Comment (jamesmbaazam) [inline R/checks.R:102]: I think this error message can be improved. For example, "If tolerance is set to 0, all distributions must have a finite `max`."

```r
> delay_opts(LogNormal(mean = 4, sd = 1), tolerance = 0)
Error in check_stan_delay(dist) : 
  All distribution passed to the model need to have a finite maximum
```

**Review by jamesmbaazam — CHANGES_REQUESTED** [review]
Thanks, Seb. This is a better fix than I suggested in the issue.

File: R/checks.R (hunk 2 above)
Comment (sbfnk) [reply R/checks.R]: Have reworded to "All distribution passed to the model need to have a finite maximum, which can be achieved either by setting `max` or non-zero `tolerance`."

**Review by jamesmbaazam — APPROVED** [review]

**Pull request merged**